plutus-core/plutus-core/src/PlutusCore/Parser/Builtin.hs

```diff
@@ -34,56 +34,68 @@ builtinFunction = lexeme $ do
             customFailure $ UnknownBuiltinFunction txt pos lBuiltin
         Just builtin -> pure builtin
 
-signedInteger :: Parser Integer
-signedInteger = Lex.signed whitespace (lexeme Lex.decimal)
-
 -- | Parser for integer constants.
-conInt :: Parser (Some (ValueOf DefaultUni))
-conInt = do
-    con::Integer <- signedInteger
-    pure $ someValue con
+conInteger :: Parser Integer
+conInteger = Lex.signed whitespace (lexeme Lex.decimal)
 
 -- | Parser for a pair of hex digits to a Word8.
 hexByte :: Parser Word8
 hexByte = do
     high <- hexDigitChar
     low <- hexDigitChar
-    return $ fromIntegral (hexDigitToInt high * 16 + hexDigitToInt low)
+    pure $ fromIntegral (hexDigitToInt high * 16 + hexDigitToInt low)
 
 -- | Parser for bytestring constants. They start with "#".
-conBS :: Parser (Some (ValueOf DefaultUni))
-conBS = do
-    _ <- char '#'
-    bytes <- Text.Megaparsec.many hexByte
-    pure $ someValue $ pack bytes
+conBS :: Parser ByteString
+conBS = lexeme . fmap pack $ char '#' *> many hexByte
 
 -- | Parser for string constants. They are wrapped in double quotes.
-conText :: Parser (Some (ValueOf DefaultUni))
-conText = do
-    con <- char '\"' *> manyTill Lex.charLiteral (char '\"')
-    pure $ someValue $ T.pack con
+conText :: Parser T.Text
+conText = lexeme . fmap T.pack $ char '\"' *> manyTill Lex.charLiteral (char '\"')
```

Comment:
One instance where we had the Whitespace invariant violated (added a Note). Not sure if I fixed all of them, but at least now we only have calls to `whitespace` in the places where they belong.